cmake/gcovr.cmake

```diff
@@ -0,0 +1,16 @@
+if(USE_GCOV)
+ 
+    SET(GCOVR gcovr)
+ 
+    SET(GCC_COVERAGE_COMPILE_FLAGS "-fprofile-arcs -ftest-coverage")
+    SET(GCC_COVERAGE_LINK_FLAGS "-lgcov")
+    SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GCC_COVERAGE_COMPILE_FLAGS}" )
```

Comment:
We are moving away from using `CMAKE_CXX_FLAGS` variables which is considered a bad practice in CMake. Instead of that, we should be using `add_compile_options` and `add_link_options`. However, the second one is only available from CMake 3.13 and therefore we keep using `CMAKE_EXE_LINKER_FLAGS` and `CMAKE_SHARED_LINKER_FLAGS` instead. You can take a look to the file `cmake/compilerFlags.cmake` to see some examples.